submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,339 @@
+        <label for="nameAZ" class="radio-lable">A - Z</label>
+        <input type="radio" id="nameZA" name="sort-name">
+        <label for="nameZA" class="radio-lable">Z - A</label>
+        <input type="radio" id="nameAll" name="sort-name">
+        <label for="nameAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+      
+      <fieldset class="filter filter-age">
+        <p>sort by age</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="age09" name="sort-age">
+        <label for="age09" class="radio-lable">young - old</label>
+        <input type="radio" id="age90" name="sort-age">
+        <label for="age90" class="radio-lable">old - young</label>
+        <input type="radio" id="ageAll" name="sort-age">
+        <label for="ageAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+
+      
+      <fieldset class="filter filter-search">
+        <p>Search user by email</p>
+        <input id="nameInput" type="text" size="30" class="searchEmail icon search">
+      </fieldset>
+    `;
+  }
+
+  renderFilters() {
+    this.filtersContainer.insertAdjacentHTML('beforeend', this.getHtmlFilter());
+  }
+}
+
+class Filter {
+  constructor(render) {
+    this.profilesContainer = document.querySelector('.profile-cards');
+    this.renderDataWithFilters = render;
+    render.renderFilters();
+    this.filtersPanel = document.querySelector('#filters');
+    this.searchInput = document.querySelector('#nameInput');
+    this.sortedUsers = [];
+    this.sortByNameTypes = {
+      inAscending: 'nameAZ',
+      inDescending: 'nameZA',
+      byDefault: 'nameAll',
+    };
+    this.sortByAgeTypes = {
+      inAscending: 'age09',
+      inDescending: 'age90',
+      byDefault: 'ageAll',
+    };
+  }
+
+  unckeckedRadioButton(nodeName) {
+    const checkedRadio = document.querySelector(
+      `input[type=radio][name=${nodeName}]:checked `
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSortMethod(selectedRadioButton) {
+    const name = selectedRadioButton.name;
+    const sortMethod = {
+      'sort-gender': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.unckeckedRadioButton('sort-age');
+        this.sortByGender(selectedRadioButton.id);
+      },
+      'sort-name': () => {
+        this.unckeckedRadioButton('sort-age');
+        this.sortByContent(
+          selectedRadioButton.id,
+          this.sortByNameTypes,
+          'name'
+        );
+      },
+      'sort-age': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.sortByContent(selectedRadioButton.id, this.sortByAgeTypes, 'age');
+      },
+    };
+    return sortMethod[name]();
+  }
+
+  startFilter() {
+    this.usersByDefault = [...this.usersDataForFilter];
+
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let usersByEmail = [];
+      let filter = target.value.toUpperCase();
```

Comment:
It is not a `filter`, it's `userInput`. Filter - is entity which filters something.

Reply:
Ok, renamed.